src/signTx.c

```diff
@@ -349,26 +382,56 @@ static void signTx_handleInit_ui_runStep()
 
 	UI_STEP_BEGIN(ctx->ui_step, this_fn);
 
-	UI_STEP(HANDLE_INIT_STEP_DISPLAY_DETAILS) {
-		char* header =
-		        (ctx->commonTxData.txSigningMode == SIGN_TX_SIGNINGMODE_MULTISIG_TX) ?
-		        "Multisig transaction" :
-		        "New transaction";
+	UI_STEP(HANDLE_INIT_STEP_PROMPT_SIGNINGMODE) {
+		char bodyTxt[SIZEOF("Pool operator") + 1 + SIZEOF("transaction?") + 1];
+		const uint32_t signingModeStrIndex = ctx->commonTxData.txSigningMode - SIGN_TX_SIGNINGMODE_ORDINARY_TX;
+		ASSERT(signingModeStrIndex < ARRAY_LEN(uiSigningModeName));
+		snprintf(bodyTxt, SIZEOF(bodyTxt), "%s transaction?", ((const char*)PIC(uiSigningModeName[signingModeStrIndex])));
+		ui_displayPrompt(
+		        "Start new",
+		        bodyTxt,
+		        this_fn,
+		        respond_with_user_reject
+		);
+	}
+
+	UI_STEP(HANDLE_INIT_STEP_DISPLAY_NETWORK_DETAILS) {
+		const bool usualNetworkId = ctx->commonTxData.networkId == MAINNET_NETWORK_ID || ctx->commonTxData.networkId == TESTNET_NETWORK_ID;
```

Comment:
```suggestion
		const bool usualNetworkId =
			ctx->commonTxData.networkId == MAINNET_NETWORK_ID ||
			ctx->commonTxData.networkId == TESTNET_NETWORK_ID;
```
The long line does not fit on my screen.